### PR TITLE
Bank data validation

### PR DESCRIPTION
**Pull request by KaiNissen**

### CODE REVIEW

File: src/Validation/DonationValidator.php [hunk 1]
```diff
@@ -27,13 +28,16 @@ public function __construct( AmountValidator $amountValidator,
 								 PersonNameValidator $nameValidator,
 								 PhysicalAddressValidator $addressValidator,
 								 AllowedValuesValidator $paymentTypeValidator,
+								 BankDataValidator $bankDataValidator,
 								 MailValidator $mailValidator ) {
+		// TODO: inject a ValidatorObject to reduce number of parameters
```
Comment (JeroenDeDauw) [inline src/Validation/DonationValidator.php]: What do you mean by this?

File: tests/Unit/Validation/BankDataValidatorTest.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+ * @license GNU GPL v2+
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
+ */
+class BankDataValidatorTest extends ValidatorTestCase {
+
+	/**
+	 * @dataProvider invalidBankDataProvider
+	 * @param string $iban
+	 * @param string $bic
+	 * @param string $bankName
```
Comment (JeroenDeDauw) [inline tests/Unit/Validation/BankDataValidatorTest.php]: Please omit param docs for test methods with data providers. I know that recent PHPStorm bitches about this (it used to not do that, I already filed an issue about it). The docs are out of date here btw :)

**Comment by JeroenDeDauw** [comment]
+1

File: src/Validation/DonationValidator.php (hunk 1 above)
Comment (KaiNissen) [inline src/Validation/DonationValidator.php]: I'm unhappy about how the list of expected parameters to create the use case object increases and thought about injecting a single container for all those validators.

**Pull request merged**